src/include/ceph_features.h

```diff
@@ -32,10 +32,13 @@
 // the bit because it ensures that we have the bit AND the marker for the
 // bit's incarnation.  this must be used in any case where the features
 // bits may include an old meaning of the bit.
+#define HAVE_DEPRECATED_FEATURE(x, name) \
+	(((x) & (DEPRECATED_CEPH_FEATUREMASK_##name)) == \
+	 (DEPRECATED_CEPH_FEATUREMASK_##name))
+
```

Comment:
I think we shouldn't do this.  What "deprecated" normally means in ceph_features.h is that, as of this release, nobody *checks* for this feature.  But this is basically make it possible to check for this feature even though it is deprecated.  I think it's going to lead to confusion/tears later down the line..

Reply:
so, what would be our solution to eventually retire features? as I understand it, we can't drop them yet because kernel clients may still be dependent on them - and that's fine -, but at some point we'll need to start dropping them (_when_ would be a good question too).

Maybe this is indeed confusing and can lead to some annoyances down the line (thus happy to drop it), but seems to me that making sure that we annotate that some of these features should be going away (or not having things depending on them) in the near future is not such a bad idea. Any thoughts on that?